### PR TITLE
NEPT-968: Allow cron to run workbench_moderation_set_state_action

### PR DESCRIPTION
**Pull request by dxvargas**

### CODE REVIEW

File: tests/features/scheduler.feature [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    And I fill in "Title" with "Next content"
+    And I fill in "publish_on[date]" with "2000-12-31"
+    And I fill in "publish_on[time]" with "23:59:59"
+    And I press the "Save" button
+    Then I should see the error message "The 'publish on' date must be in the future"
+    And I change the variable "scheduler_publish_past_date_page" to "schedule"
+    And I press the "Save" button
+    Then I should see the message "This post is unpublished and will be published 2000-12-31 23:59:59."
+    And I should see the text "Revision state: Draft"
+    And I am on "admin/config/system/cron"
```
Comment (voidtek) [inline tests/features/scheduler.feature]: We can use the step "@Given I run cron" provided by DrupalContext.php

File: tests/features/scheduler.feature [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    # And I click "Scheduling options"
+    And I fill in "unpublish_on[date]" with "2000-12-31"
+    And I fill in "unpublish_on[time]" with "23:59:59"
+    And I press the "Save" button
+    Then I should see the error message "The 'unpublish on' date must be in the future"
+    When I fill in "unpublish_on[date]" with "2100-12-31"
+    And I press the "Save" button
+    Then I should see the text "Revision state: Published"
+    When I make dates from "unpublish_on" in db table "scheduler" to be in past
+    And I am on "admin/config/system/cron_en"
```
Comment (voidtek) [inline tests/features/scheduler.feature]: We can use the step "@given I run cron" provided by DrupalContext.php

File: tests/features/scheduler.feature [hunk 3]
```diff
@@ -0,0 +1,47 @@
+@api @run
```
Comment (voidtek) [inline tests/features/scheduler.feature]: What is the purpose of "@run"?

File: tests/src/Context/DrupalContext.php [hunk 4]
```diff
@@ -209,4 +209,15 @@ public function theFollowingContentsUsingForWysiwygFields($text_format, TableNod
     return $nodes;
   }
 
+  /**
+   * Update dates in db table field to be in the past.
+   *
+   * @When I make dates from :field in db table :table to be in past
+   */
+  public function iMakeDatesFromFieldInDbTableToBeInPast($field, $table) {
+    db_update($table)
```
Comment (voidtek) [inline tests/src/Context/DrupalContext.php]: This update changes all items on the scheduler's table. Maybe, we can specify a item.

File: tests/src/Context/DrupalContext.php [hunk 5]
```diff
@@ -209,4 +209,15 @@ public function theFollowingContentsUsingForWysiwygFields($text_format, TableNod
...
+      ->fields(array($field => time() - 1))
```
Comment (voidtek) [inline tests/src/Context/DrupalContext.php]: We do not check if the field exists in the table.

File: tests/features/scheduler.feature [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+Feature: Scheduler features
+  In order to moderate contents
+  As an administrator
+  I want to be able to schedule state transitions for contents
+
+  Background:
+    Given I am logged in as a user with the 'administrator' role
+    And I change the variable "scheduler_publish_enable_page" to 1
```
Comment (voidtek) [inline tests/features/scheduler.feature]: With a clean platform, this variable is already equals to 1. I think that this step is not useful.

File: tests/features/scheduler.feature [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    And I change the variable "scheduler_unpublish_enable_page" to 1
```
Comment (voidtek) [inline tests/features/scheduler.feature]: With a clean platform, this variable is already equals to 1. I think that this step is not useful.

File: tests/features/scheduler.feature (hunk 3 above)
Comment (dxvargas) [inline tests/features/scheduler.feature]: It should have been deleted.

**Comment by voidtek** [comment]
:+1:

File: resources/multisite_drupal_standard.make [hunk 8]
```diff
@@ -770,6 +770,10 @@ projects[workbench_moderation][patch][] = https://www.drupal.org/files/issues/no
 ; https://www.drupal.org/node/1084436
 ; https://webgate.ec.europa.eu/CITnet/jira/browse/NEPT-592
 projects[workbench_moderation][patch][] = patches/workbench_moderation-002-1084436-2645622-merge.patch
+; Allow cron to run workbench_moderation_set_state_action
+; https://www.drupal.org/node/2476489
+; https://webgate.ec.europa.eu/CITnet/jira/browse/NEPT-968
+projects[workbench_moderation][patch][] = https://www.drupal.org/files/issues/workbench_moderation-reimplement-force_transaction-247648907.patch
```
Comment (gillesdeudon) [inline resources/multisite_drupal_standard.make]: It is not the latest patch on the Drupal issue. See the PR, it uses https://www.drupal.org/files/issues/force_transition-2476489-12.patch.
Why od we have this difference? Is it a mistake or done on purpose?

**Comment by gillesdeudon** [comment]
:+1:

**Comment by voidtek** [comment]
:+1:

**Pull request merged**